AsyncQueue.ts

```diff
@@ -0,0 +1,30 @@
+import { AsyncQueueConsumer } from "./AsyncQueueConsumer";
+
+export class AsyncQueue {
+  private queue: Array<Array<any>> = [];
+  private consumer: AsyncQueueConsumer;
+
+  constructor(callback: Function) {
```

Comment:
I think a better name here is handler.
```suggestion
  constructor(handler: Function) {
```